libraries/chain/include/scorum/chain/database/block_tasks/comments_cashout_impl.hpp

```diff
@@ -56,72 +56,24 @@ class process_comments_cashout_impl
         });
     }
 
-    template <typename FundService> void reward(FundService& fund_service, const comment_refs_type& comments)
-    {
-        using fund_object_type = typename FundService::object_type;
-
-        const auto& rf = fund_service.get();
-
-        if (rf.activity_reward_balance.amount < 1 || comments.empty())
-            return;
-
-        shares_vector_type total_rshares = get_total_rshares(comments);
-        fc::uint128_t total_claims
-            = rewards_math::calculate_total_claims(rf.recent_claims, rf.author_reward_curve, total_rshares);
-
-        struct comment_reward
-        {
-            share_type fund; // reward accrued from fund
-            share_type commenting; // reward accrued from children comments
-        };
-
-        std::map<account_name_type, comment_reward> comment_rewards;
-
-        for (const comment_object& comment : comments)
-        {
-            share_type payout = rewards_math::calculate_payout(
-                comment.net_rshares, total_claims, rf.activity_reward_balance.amount, rf.author_reward_curve,
-                comment.max_accepted_payout.amount, SCORUM_MIN_COMMENT_PAYOUT_SHARE);
-
-            comment_rewards.emplace(comment.author, comment_reward{ payout, 0 });
-        }
-
-        asset_symbol_type reward_symbol = rf.activity_reward_balance.symbol();
-        asset total_reward = asset(0, reward_symbol);
-
-        // newest, with bigger depth comments first
-        comment_refs_type comments_with_parents = collect_parents(comments);
-
-        for (const comment_object& comment : comments_with_parents)
-        {
-            const comment_reward& reward = comment_rewards[comment.author];
+    template <typename TFundService> void reward(TFundService& fund_service, const comment_refs_type& comments);
```

Comment:
Why did you move ::reward and other methods to CPP? It greatly complicates the review. 
Do that in separate pull request in future.